src/main/java/demurebot/storage/Storage.java

```diff
@@ -0,0 +1,190 @@
+import demurebot.task.Deadline;
+import demurebot.task.Event;
+import demurebot.task.Task;
+import demurebot.task.TaskList;
+import demurebot.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the storage of tasks in the DemureBot application.
+ * A Storage object is responsible for loading tasks from a file and saving tasks to a file.
+ */
+public class Storage {
+    private final ArrayList<Task> list;
+
+    /**
+     * Constructs a new Storage object with an empty list of tasks.
+     */
+    public Storage() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Checks if the folder exists.
+     *
+     * @param folderPath Path of folder containing the file with the saved data
+     */
+    public void checkFolder(String folderPath) {
+        File folder = new File(folderPath);
+        if (!folder.exists()) {
+            boolean isFolderCreated = folder.mkdirs();
+            if (!isFolderCreated) {
+                System.out.println("Failed to create directory: " + folderPath);
+            }
+        }
+    }
+
+    /**
+     * Returns a Boolean indicating if the file exists.
+     *
+     * @param filePath Path of file containing saved data
+     * @return Boolean indicating if the file exists
+     */
+    public boolean checkFile(String filePath) {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                boolean isFileCreated = file.createNewFile();
+                if (!isFileCreated) {
+                    System.out.println("Failed to create file: " + filePath);
+                }
+                return false;
+            } catch (IOException e) {
+                System.out.println("Error creating file: " + e.getMessage());
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Loads the saved data from the file.
+     *
+     * @param filePath Path of file containing saved data
+     * @return List of tasks loaded from the file
+     * @throws IOException        If an I/O error occurs
+     * @throws DemureBotException If the saved task has invalid format
+     */
+    public ArrayList<Task> load(String filePath) throws IOException, DemureBotException {
+        Scanner fileScanner = new Scanner(new File(filePath));
+        while (fileScanner.hasNext()) {
+            String line = fileScanner.nextLine();
+            Task task = getTask(line);
+            list.add(task);
+        }
+        return list;
+    }
+
+    /**
+     * Returns a demure.Task created from saved data.
+     *
+     * @param line Line of text read from saved data.
+     * @return demure.Task created from saved data.
+     * @throws DemureBotException If the saved task has invalid format.
+     */
+    private Task getTask(String line) throws DemureBotException {
+        String[] split = line.split(" \\| ");
+        validateTaskFormat(split, 3);
```

Comment:
Should the ```validateTaskFormat(split, 3)``` be inside the ```case "T"```, similar to how deadline and event are validated first before creation?